drjt/MyDate/lib/MyDate.pm

```diff
@@ -0,0 +1,169 @@
+package MyDate;
+
+use 5.010;
+use strict;
+use warnings;
+
+use Carp qw( croak );
+use Time::Piece;
+
+=head1 NAME
+
+MyDate - The great new MyDate!
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+
+=head1 SYNOPSIS
+
+Quick summary of what the module does.
+
+Perhaps a little code snippet.
+
+    use MyDate;
+
+    my $foo = MyDate->new();
+    ...
+
+=head1 EXPORT
+
+A list of functions that can be exported.  You can delete this section
+if you don't export anything, such as for a purely object-oriented module.
+
+=head1 SUBROUTINES/METHODS
+
+=head2 AUTOLOAD
+
+=cut
+
+sub AUTOLOAD {
+  our $AUTOLOAD;
+  my $method = $AUTOLOAD =~ s/.*:://r;
+  if ( $method eq 'date' ) {
+    eval {
+      sub date {
+        return localtime()->date();
+      }
+    };
+    die $@ if $@;
+    goto &date;
+  }
+  elsif ( $method eq 'month' ) {
+    eval {
```

Comment:
Didn't really need these evals here. Even if Time::Piece dies, you're just repeating the error messages in a different place and masking where the issue is arising.